source/extensions/filters/http/api_key_auth/api_key_auth.cc

```diff
@@ -135,6 +167,22 @@ Http::FilterHeadersStatus ApiKeyAuthFilter::decodeHeaders(Http::RequestHeaderMap
     }
   }
 
+  if (forwarding.has_value()) {
+    const Http::LowerCaseString& header_name = forwarding->getHeaderName();
+
+    if (!header_name.get().empty()) {
+      const std::string& client = credential->second;
+      // Add the authenticated client to the request headers.
+      headers.setCopy(header_name, client);
+      ENVOY_LOG(debug, "Inserted client ID {} into header {}", client, header_name.get());
```

Comment:
I personally think this log is redudant because if debug logging is enabled, the user will see the header in the log of router filter.